realm/realm-library/src/main/java/io/realm/HandlerController.java

```diff
@@ -367,7 +372,13 @@ private void completedAsyncRealmResults(QueryUpdateTask.Result result) {
                     if (!realmResults.isLoaded()) {
                         RealmLog.d("[COMPLETED_ASYNC_REALM_RESULTS "+ weakRealmResults + "] , realm:"+ HandlerController.this + " same versions, using results (RealmResults is not loaded)");
                         // swap pointer
-                        realmResults.swapTableViewPointer(result.updatedTableViews.get(weakRealmResults));
+                        try {
+                            realmResults.swapTableViewPointer(result.updatedTableViews.get(weakRealmResults));
+                        } catch (BadVersionException e) {
```

Comment:
This _can't_ happen, since the CallerThread.version == BgThread.version & the caller thread is holding an open SharedGroup using this version.